### PR TITLE
Adds new delete command to cob_index.

### PR DESCRIPTION
**Pull request by dkinzer**
REF BL-1125

Usage.

cob_index delete url/or path to oai xml file with deleted records.

### depends on:
- [ ] tulibraries/tulflow#49

### CODE REVIEW

File: lib/cob_index.rb [hunk 1]
```diff
@@ -2,14 +2,29 @@
...
-      indexer.process(StringIO.new(open(ARGV[0]).read))
+      indexer.process(StringIO.new(marc_xml))
+    end
+
+
+    def self.delete(commit: false, xml: "")
+      settings =  {
+        "nokogiri.each_record_xpath" => "/oai:OAI-PMH/oai:ListRecords/oai:record",
+        "nokogiri.namespaces" => { "oai" => "http://www.openarchives.org/OAI/2.0/" },
+        "solr_writer.commit_on_close" => commit
```
Comment (sensei100) [inline lib/cob_index.rb]: @dkinzer Will this just end up being one commit at the end of the task?

File: lib/cob_index/default_config.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+module CobIndex
+  module  DefaultConfig
+    # Wrapper for our default indexer settings so that we can share them
+    # across commands.
+    def self.indexer_settings
+      solr_url = if File.exist? "config/blacklight.yml"
+                      solr_config = YAML.load_file("config/blacklight.yml")[(ENV["RAILS_ENV"] || "development")]
+                      ERB.new(solr_config["url"]).result
+                    else
+                      ENV["SOLR_URL"] || "http://localhost:8993/solr/blackligh"
```
Comment (sensei100) [inline lib/cob_index/default_config.rb]:
```suggestion
                      ENV["SOLR_URL"] || "http://localhost:8993/solr/blacklight"
```

File: lib/cob_index.rb (hunk 1 above)
Comment (dkinzer) [reply lib/cob_index.rb]: No, the default is not to commit so here commit = false.

That  is there so we can pass `cob_index delete --commit file` which is useful for local development purposes.

File: lib/cob_index/default_config.rb (hunk 2 above)
Comment (dkinzer) [reply lib/cob_index/default_config.rb]: oh, that shouldn't even be there... just added to my local for testing purposes 🤦‍♂

**Pull request merged**